packages/scene-composer/src/hooks/useBindingData.ts

```diff
@@ -38,39 +38,41 @@ const useBindingData = (
     }
 
     const providers = queries.map((query, index) => {
-      if (query) {
-        const provider = query.build(sceneComposerId, {
-          viewport: viewport,
-          settings: {
-            // only support default settings for now until when customization is needed
-            fetchFromStartToEnd: true,
-          },
-        });
+      if (!query) {
+        return undefined;
+      }
 
-        provider.subscribe({
-          // TODO: support static data
-          next: (results: TimeSeriesData[]) => {
-            if (isEmpty(results.at(0)?.dataStreams)) {
-              log?.info('No data returned');
-              return;
-            }
+      const provider = query.build(sceneComposerId, {
+        viewport: viewport,
+        settings: {
+          // only support default settings for now until when customization is needed
+          fetchFromStartToEnd: true,
+        },
+      });
 
-            if (results.length > 1 || results[0].dataStreams.length > 1) {
-              log?.warn('Multiple data returned, only use the first one');
-            }
+      provider.subscribe({
+        // TODO: support static data
+        next: (results: TimeSeriesData[]) => {
+          if (isEmpty(results.at(0)?.dataStreams)) {
```

Comment:
You can directly check results[0].dataStreams?

Reply:
both results and results[0].dataStreams can be empty array, so I need to check bth. if you are talking about [0] vs at(0), they should be the same?

Reply:
Yeah bit confusing as you used some place results.at(0) and in some results[0] both.

Reply:
using at(0) force me to handle the undefined case by throw compiler error, but `results[0].dataStreams` will not throw error when building, but will throw runtime error when results is empty. for the code below, since I've checked they are not empty here, I can just access [0] and impossible to have undefined